$OpenBSD: patch-src_iconloader_qticonloader_cpp,v 1.5 2012/01/02 08:35:48 dcoppa Exp $

Use the gnome icon theme unconditionally: fixes a problem with minitube
GUI not having icons under fvwm/cwm/kde3/...

--- src/iconloader/qticonloader.cpp.orig	Fri Oct 28 22:38:22 2011
+++ src/iconloader/qticonloader.cpp	Fri Dec 30 11:25:56 2011
@@ -107,7 +107,6 @@ Q_GLOBAL_STATIC(QtIconLoaderImplementation, iconLoader
     QIcon icon;
 
 #ifdef Q_WS_X11
-#if QT_VERSION < 0x040600
     QString pngExtension(QLatin1String(".png"));
     QList<int> iconSizes;
     iconSizes << 16 << 22 << 24 << 32 << 48;
@@ -115,9 +114,6 @@ Q_GLOBAL_STATIC(QtIconLoaderImplementation, iconLoader
         icon.addPixmap(iconLoaderInstance()->findIcon(size, name));
     }
 #else
-    icon = QIcon::fromTheme(name);
-#endif
-#else
         icon = Extra::getIcon(name);
 #endif
 
@@ -178,10 +174,14 @@ void QtIconLoaderImplementation::lookupIconTheme() con
 #ifdef Q_WS_X11
     QString dataDirs = QFile::decodeName(getenv("XDG_DATA_DIRS"));
     if (dataDirs.isEmpty())
-        dataDirs = QLatin1String("/usr/local/share/:/usr/share/");
+        dataDirs = QLatin1String("${LOCALBASE}/share/");
     
     dataDirs.prepend(QDir::homePath() + QLatin1String("/:"));
     iconDirs = dataDirs.split(QLatin1Char(':'));
+
+    if (themeName.isEmpty())
+	themeName = QLatin1String("gnome");
+    return;
     
     // If we are running GNOME we resolve and use GConf. In all other
     // cases we currently use the KDE icon theme
